db/db_impl_open.cc

```diff
@@ -369,6 +369,10 @@ Status DBImpl::Recover(
     default_cf_internal_stats_ = default_cf_handle_->cfd()->internal_stats();
     single_column_family_mode_ =
         versions_->GetColumnFamilySet()->NumberOfColumnFamilies() == 1;
+    uint64_t min_log_num = kMaxSequenceNumber;
+    for (auto cfd : *versions_->GetColumnFamilySet()) {
+      min_log_num = std::min(min_log_num, cfd->GetLogNumber());
```

Comment:
This is not the expected behavior. We want to record in the MANIFEST whenever we deleted a log file. The decision of when to delete a log file is complicated and  you do not need to get involved with that. But whenever we set a file for deletion, you can record it in the manifest, and later during recovery use that number to see which log files you should skip.